src/services/organizeImports.ts

```diff
@@ -92,7 +92,7 @@ namespace ts.OrganizeImports {
     function removeUnusedImports(oldImports: ReadonlyArray<ImportDeclaration>, sourceFile: SourceFile, program: Program) {
         const typeChecker = program.getTypeChecker();
         const jsxNamespace = typeChecker.getJsxNamespace();
-        const jsxContext = sourceFile.languageVariant === LanguageVariant.JSX && program.getCompilerOptions().jsx;
+        const jsxContext = sourceFile.languageVariant === LanguageVariant.JSX;

```

Comment:
i wounder if a better check here is `sourceFile.transformFlags & TransformFlags.ContainsJsx !== 0`.. this way we only keep it if we know at least one JSX element exist..